src/Nox.Reference.VatNumbers/Extension/StringExtensions.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Nox.Reference.VatNumbers.Constants;
+using Nox.Reference.VatNumbers.Models;
+using System.Text;
+
+namespace Nox.Reference.VatNumbers.Extension
+{
+    internal static class StringExtensions
+    {
+        public static void ValidateCustomChecksum(this string vatNumber, ValidationResult result, Func<string, ValidationResult, bool> checksumFunc)
```

Comment:
I'd rather get validation errors than fill  result.ValidationErrors inside method

Reply:
Changed it